fission-core/src/capabilities/fission.rs

```diff
@@ -211,65 +214,166 @@ mod tests {
     #[test_log::test]
     #[ignore]
     fn test_powerbox_ucan_resource() -> TestResult {
-        let store = InMemoryStore::<RawCodec>::default();
-        let did_verifier_map = DidVerifierMap::default();
+        // let store:  = ucan::delegation::store::MemoryStore::default();
+
+        let server_sk = ed25519_dalek::SigningKey::generate(&mut thread_rng());
+        let server_signer =
+            ucan::did::preset::Signer::Key(ucan::did::key::Signer::EdDsa(server_sk));
+        let server_ed_did_key = EdDidKey::new(server_sk);
+
+        let server = ucan::did::preset::Verifier::Key(ucan::did::key::Verifier::EdDsa(
+            server_sk.verifying_key(),
+        ));
 
-        let server = &EdDidKey::generate();
-        let device = &EdDidKey::generate();
+        let device = ucan::did::preset::Verifier::Key(ucan::did::key::Verifier::EdDsa(
+            ed25519_dalek::SigningKey::generate(&mut thread_rng()).verifying_key(),
+        ));
+
+        // FIXME perhaps add this back upstream as a named const
+        let varsig_header = ucan::crypto::varsig::header::EdDsaHeader {
+            codec: libipld::cbor::DagCborCodec,
+        };
+
+
+        // 1.               account -*-> server
+        // 2.                            server -a-> device
+        // 3.  dnslink -d-> account
+        // 4. [dnslink -d-> account -*-> server -a-> device]
```

Comment:
We... may have a problem with powerlines where they help with the simple case but nothing more complex.

If I'm reading the scenario that you put together correctly, it looks something like this:

```
1.               account -*-> server
2.               account -*-------------> device
3.  dnslink -d-> account
4. [dnslink -d-> account -*-------------> device]
```

Which powerlines handle perfectly 👍 

```rust
// 2.
delegation::Payload {
    subject: None,
    issuer: account_did,
    audience: device_did,
    command: "/",
    policy: vec![],
   // ...


// 3.
delegation::Payload {
    subject: Some(dnslink_did),
    issuer: dnslink_did,
    audience: account_did,
    command: "/",
    policy: vec![],
   // ...
}}

// 4.
invocation::Payload {
    subject: dnslink_did,
    issuer: device_did,
    prf: ["bafy...2", "bafy...3"]
   // ...
}}
```

...but that doesn't help if the device gets revoked/lost. We need to either put a server in the middle, or reissue UCANs by keeping a signing key around (which would be nice to avoid). The problem with using a `subject = None` from the server is that you're granting everything delegated to the _server_, which we obviously don't want.

```
1.               account -*-> server
2.                            server -a-> device
3.  dnslink -d-> account
4. [dnslink -d-> account -*-> server -a-> device]
```

This is much closer to the original design with a `Proxy` ability...

```rust
pub struct Proxy<DID: Did, T> {
    pub subject: DID,
    pub proxy: T,
    pub prf: Vec<Cid>
}
```

...which totally works iff you understand the special proxy type.

Anyhow, just something to ponder. I don't think we can wedge this into the core type, but worth putting some cycles into.

Reply:
> We need to either put a server in the middle, or reissue UCANs by keeping a signing key around (which would be nice to avoid). The problem with using a subject = None from the server is that you're granting everything delegated to the server, which we obviously don't want.

Yeah, the idea was to put the server in the middle, and yes, that would grant everything delegated to the server.

I was thinking of two ways forward:
- In the future, implement a mode where the user 'owns' their key, by means of BLS multisig + server cosigner n' stuff.
- In the future, implement multiple server keys in a hierarchy, where the 'root' key is the one that gets full control and is a BLS key.

Reply:
> implement multiple server keys in a hierarchy

Just because it's unclear in the above text: is this the version in the design docs, or a new design that relies on BLS features?

Reply:
The version from the design docs :)